### PR TITLE
Fix crash when providing None to mode arg of open()

### LINKED ISSUE #5731: Crash with argument of type 'NoneType' is not iterable
### Bug description

PyLint crashes when running on Plumbum. (it continues on other files just fine, though). Pretty sure it's choking on this:

https://github.com/tomerfiliba/plumbum/blob/bd29e7f21b1a9856d81553f149c68d07fbc38ce9/plumbum/commands/base.py#L474-L475

### Configuration

_No response_

### Command used

```shell
python -m pip install . paramiko pylint ipython
pylint plumbum

# (inside of a local noxfile)
```

### Pylint output

```pytb
Traceback (most recent call last):
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1034, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1069, in _check_file
    check_astroid_module(ast_node)
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1203, in check_astroid_module
    retval = self._check_astroid_module(
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/lint/pylinter.py", line 1250, in _check_astroid_module
    walker.walk(node)
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  [Previous line repeated 2 more times]
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/utils/ast_walker.py", line 72, in walk
    callback(astroid)
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/checkers/stdlib.py", line 534, in visit_call
    self._check_open_encoded(node, inferred.root().name)
  File "/Users/henryschreiner/git/software/plumbum/.nox/pylint/lib/python3.10/site-packages/pylint/checkers/stdlib.py", line 633, in _check_open_encoded
    and "b" not in mode_arg.value
TypeError: argument of type 'NoneType' is not iterable
```

### Expected behavior

No crash?

### Pylint version

```shell
pylint 2022.1.7
```

### OS / Environment

macOS, Python 3.10. Targeting 3.6.

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #5731

### CODE REVIEW

File: tests/functional/u/unspecified_encoding_py38.txt [hunk 1]
```diff
@@ -27,3 +27,5 @@ unspecified-encoding:149:0:149:23::Using open without explicitly specifying an e
 unspecified-encoding:152:0:152:28::Using open without explicitly specifying an encoding:UNDEFINED
 unspecified-encoding:155:0:155:26::Using open without explicitly specifying an encoding:UNDEFINED
 unspecified-encoding:158:0:158:35::Using open without explicitly specifying an encoding:UNDEFINED
+bad-open-mode:161:0:161:25::"""%s"" is not a valid mode for open.":UNDEFINED
```
Comment (jacobtylerwalls) [inline tests/functional/u/unspecified_encoding_py38.txt:30]: Ouch, that's not a good message! 🤕 

```
a.py:1:0: W1501: "%s" is not a valid mode for open. (bad-open-mode)
```

Track separately?

Comment (DanielNoord) [reply tests/functional/u/unspecified_encoding_py38.txt:30]: Oh wow, I missed that. Hmm, yeah. Should probably be trivial to fix, but should indeed go in a different issue/PR.

**Comment by henryiii** [comment]
I can verify that this solves the crash (with the same error message). :)

```diff
-    session.install(".", "paramiko", "pylint", "ipython")
+    session.install(".", "paramiko", "ipython")
+    session.install("git+https://github.com/jacobtylerwalls/pylint@mode-crash")
```

```
************* Module plumbum.commands.base
...
plumbum/commands/base.py:476:37: W1501: "%s" is not a valid mode for open. (bad-open-mode)
...
```

**Comment by henryiii** [comment]
PS: I am also fixing the default in my code from `MODE = None # type: str` to `MODE: str = "w"`, which I expect will fix the error from my side too. Thought I'd guess that `None` technically does work for mode.

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

**Comment by jacobtylerwalls** [comment]
Thanks for testing!

> I'd guess that None technically does work for mode.

I wasn't sure myself, so I headed over to a shell:

```python
>>> f = open(__name__, mode=None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: open() argument 'mode' must be str, not None
```

**Comment by Pierre-Sassoulas** [comment]
Thanks for checking the fix @henryiii, and thank you for the super fast fix @jacobtylerwalls !

**Comment by henryiii** [comment]
THat's weird, must never have hit the default. Ahh, it's always subclassed (this should have been an abstract property, probably, then). Okay, that's why it was never hit.

**Pull request merged**